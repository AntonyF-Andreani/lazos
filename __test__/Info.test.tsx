import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom";

import InfoSection, { InfoSectionProps } from "sections/Info";

const defaultProps: InfoSectionProps = {
  button_link: "url",
  button_text: "text",
  text: "description",
};

describe("<InfoSection />", () => {
  it("should render without crash", () => {
    const { asFragment } = render(<InfoSection {...defaultProps} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("should have the correct props rendered", () => {
    const { getByRole, container } = render(<InfoSection {...defaultProps} />);

    const button = getByRole("link");

    expect(button.hasAttribute("href")).toBeTruthy();
    expect(button.getAttribute("href")).toBe(defaultProps.button_link);

    expect(container).toHaveTextContent("text");
    expect(container).toHaveTextContent("description");
  });
});
