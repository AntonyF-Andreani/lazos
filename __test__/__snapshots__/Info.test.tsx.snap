// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InfoSection /> should render without crash 1`] = `
<DocumentFragment>
  <section
    class="MuiBox-root css-0"
  >
    <div
      class="containerInfoSection MuiBox-root css-0"
      max-maxwidth="1000px"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
      >
        description
      </p>
      <a
        class="MuiButtonBase-root MuiButton-root root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-disableElevation css-bb99s9-MuiButtonBase-root-MuiButton-root"
        href="url"
        tabindex="0"
      >
        <h6
          class="MuiTypography-root MuiTypography-subtitle2 css-ati914-MuiTypography-root"
        >
          text
        </h6>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </a>
    </div>
  </section>
</DocumentFragment>
`;
